### PR TITLE
feat: added glue code for dismissing an app

### PR DESCRIPTION
**Pull request by anjalimukundan**

### CODE REVIEW

File: cypress/support/step_definitions/fireboltCalls.js [hunk 1]
```diff
@@ -365,3 +365,125 @@ Given('device is rebooted', () => {
...
+    Cypress.env('app_metadata') &&
+    Cypress.env('app_metadata')[appId] &&
+    Cypress.env('app_metadata')[appId].defaultKeyPressSequence
+  ) {
+    KeyPressSequence = Cypress.env('app_metadata')[appId].defaultKeyPressSequence;
+  } else if (Cypress.env('app_metadata') && Cypress.env('app_metadata').defaultKeyPressSequence) {
+    KeyPressSequence = Cypress.env('app_metadata').defaultKeyPressSequence;
+  } else {
+    throw new Error('No KeyPressSequence ');
+  }
```
Comment (neeradanelxsi) [inline cypress/support/step_definitions/fireboltCalls.js:396]: Log should be descriptive. how to fix if there is no kepress sequence

File: cypress/support/step_definitions/fireboltCalls.js [hunk 2]
```diff
@@ -365,3 +365,125 @@ Given('device is rebooted', () => {
...
+        );
+      }
+      const contentObject = {
+        logs: logs,
+        content: {
+          type: {
+            methods: {
+              'lifecycle.close': [],
+            },
+          },
```
Comment (neeradanelxsi) [inline cypress/support/step_definitions/fireboltCalls.js:482]: what happens for badger here ?

File: cypress/support/step_definitions/fireboltCalls.js [hunk 3]
```diff
@@ -365,3 +365,125 @@ Given('device is rebooted', () => {
...
+        params: {
+          validations: [
+            {
+              type: 'image',
+              label: 'home',
+              confidence: 50,
+            },
+          ],
+        },
+      };
```
Comment (neeradanelxsi) [inline cypress/support/step_definitions/fireboltCalls.js:451]: I believe the ask was to create a separate validation object for deeplink and use it here, please check grooming video

File: cypress/support/step_definitions/fireboltCalls.js (hunk 2 above)
Comment (neeradanelxsi) [reply cypress/support/step_definitions/fireboltCalls.js:482]: why are we hardcoding the content object/validation objects in FCS ?

File: cypress/support/step_definitions/fireboltCalls.js (hunk 1 above)
Comment (anjalimukundan) [reply cypress/support/step_definitions/fireboltCalls.js:396]: updated the log

File: cypress/support/step_definitions/fireboltCalls.js (hunk 3 above)
Comment (anjalimukundan) [reply cypress/support/step_definitions/fireboltCalls.js:447]: created deeplink.js file for both screenshot and interactionLogs

File: cypress/support/step_definitions/fireboltCalls.js [hunk 4]
```diff
@@ -365,3 +365,121 @@ Given('device is rebooted', () => {
...
+      if (Object.keys(response).length === 0) {
+        fireLog.info(
+          `State validation successful: Current state of ${appId} app is ${responseString} as expected`
+        );
+      } else {
+        fireLog.fail(`${appId} app is not dismissed. Response :${responseString}`);
+      }
+    })
+    .then(() => {
+      // screenShot validation
```
Comment (neeradanelxsi) [inline cypress/support/step_definitions/fireboltCalls.js:440]: when is the firebolt interactions enabled when executing deeplink ? like when does it start collection

File: cypress/support/step_definitions/fireboltCalls.js [hunk 5]
```diff
@@ -365,3 +365,121 @@ Given('device is rebooted', () => {
...
+      fireLog.info('Started Screenshot validation');
+      const requestMapForScreenShotValidation = {
+        method: CONSTANTS.REQUEST_OVERRIDE_CALLS.SCREENSHOT,
```
Comment (neeradanelxsi) [inline cypress/support/step_definitions/fireboltCalls.js:443]: there is also another question about dismiss right? about native apps/hybrid. are we covering that here ? else that need to be called out

File: cypress/support/step_definitions/fireboltCalls.js (hunk 3 above)
Comment (neeradanelxsi) [reply cypress/support/step_definitions/fireboltCalls.js:447]: the feature is dismiss, so we need to create for dismiss

File: cypress/support/step_definitions/fireboltCalls.js (hunk 2 above)
Comment (anjalimukundan) [reply cypress/support/step_definitions/fireboltCalls.js]: moved to fixtures

File: cypress/support/step_definitions/fireboltCalls.js (hunk 4 above)
Comment (anjalimukundan) [reply cypress/support/step_definitions/fireboltCalls.js:440]: in the beforeAll
image

File: cypress/support/step_definitions/fireboltCalls.js (hunk 5 above)
Comment (anjalimukundan) [reply cypress/support/step_definitions/fireboltCalls.js:435]: added changes to fetch apptype and get data

File: cypress/support/step_definitions/fireboltCalls.js (hunk 3 above)
Comment (anjalimukundan) [reply cypress/support/step_definitions/fireboltCalls.js:439]: updated as Dimiss

**Review by jbigel — COMMENTED** [review]
looks good, but waiting to approve until deeplink has been merged

File: cypress/support/step_definitions/fireboltCalls.js (hunk 1 above)
Comment (neeradanelxsi) [reply cypress/support/step_definitions/fireboltCalls.js:396]: app_metadata should be in constant for deeplink and dismiss

Comment (neeradanelxsi) [reply cypress/support/step_definitions/fireboltCalls.js:396]: throw new Error(`Expected KeyPressSequence was not found for ${appId} in app_metadata.json. More details on app_metadata present <link to app metadatajson template and readme>`); - 

add a template for app_metadata.json in FCS with all dummy values

File: cypress/support/step_definitions/fireboltCalls.js [hunk 6]
```diff
@@ -365,3 +365,98 @@ Given('device is rebooted', () => {
...
+  const appId = Cypress.env(CONSTANTS.CURRENT_APP_ID);
+  const requestMapForGetAppState = {
+    method: CONSTANTS.REQUEST_OVERRIDE_CALLS.GETAPPSTATE,
+    params: appId,
+  };
+  fireLog.info(
+    `Sending request to fetch ${appId} app state: ${JSON.stringify(requestMapForGetAppState)}`
+  );
+  cy.sendMessagetoPlatforms(requestMapForGetAppState)
+    .then((response) => {
```
Comment (neeradanelxsi) [inline cypress/support/step_definitions/fireboltCalls.js:421]: comments against each if/else condition

File: cypress/support/step_definitions/fireboltCalls.js [hunk 7]
```diff
@@ -365,3 +365,98 @@ Given('device is rebooted', () => {
...
+          response.visibilityState.toUpperCase() === CONSTANTS.VISIBLE
+        ) {
+          cy.log(
+            `State validation successful: Current state of ${appId} app is ${responseString} as expected`
+          );
+        } else {
+          fireLog.fail(`${appId} app is not dismissed. Response :${responseString}`);
+        }
+      }
+    })
```
Comment (neeradanelxsi) [inline cypress/support/step_definitions/fireboltCalls.js:442]: else condition for line 423 is missing

File: cypress/support/step_definitions/fireboltCalls.js (hunk 1 above)
Comment (anjalimukundan) [reply cypress/support/step_definitions/fireboltCalls.js:406]: added a template app_metadata.json

File: cypress/support/step_definitions/fireboltCalls.js (hunk 6 above)
Comment (anjalimukundan) [reply cypress/support/step_definitions/fireboltCalls.js:431]: added comments

File: cypress/support/step_definitions/fireboltCalls.js (hunk 7 above)
Comment (anjalimukundan) [reply cypress/support/step_definitions/fireboltCalls.js:459]: removed the main if condition and added sub if-else

File: cypress/support/step_definitions/fireboltCalls.js [hunk 8]
```diff
@@ -365,3 +365,115 @@ Given('device is rebooted', () => {
...
+      } else if (
+        // Check if appState is missing, visibilityState exists, and visibilityState is VISIBLE
+        response &&
+        !response.appState &&
+        response.visibilityState &&
+        response.visibilityState.toUpperCase() === CONSTANTS.VISIBLE
+      ) {
+        fireLog.info(
+          `State validation successful: Current state of ${appId} app is ${responseString} as expected`
+        );
```
Comment (neeradanelxsi) [inline cypress/support/step_definitions/fireboltCalls.js:454]: line 453 is not correct. we are not getting any app state and we are not validating app state here

File: cypress/support/step_definitions/fireboltCalls.js [hunk 9]
```diff
@@ -365,3 +365,115 @@ Given('device is rebooted', () => {
...
+  };
+  fireLog.info(
+    `Sending request to fetch ${appId} app state: ${JSON.stringify(requestMapForGetAppState)}`
+  );
+  cy.sendMessagetoPlatforms(requestMapForGetAppState)
+    .then((response) => {
+      const responseString = JSON.stringify(response);
+      if (
+        // Check if response exists and contains appState, visibilityState
+        // and if appState is INACTIVE and visibilityState is VISIBLE
```
Comment (neeradanelxsi) [inline cypress/support/step_definitions/fireboltCalls.js:435]: give a generic comment on line 435. what is visibility state ?

File: cypress/support/step_definitions/fireboltCalls.js (hunk 8 above)
Comment (anjalimukundan) [reply cypress/support/step_definitions/fireboltCalls.js:454]: updated the logs

File: cypress/support/step_definitions/fireboltCalls.js (hunk 9 above)
Comment (anjalimukundan) [reply cypress/support/step_definitions/fireboltCalls.js]: updated

File: cypress/support/cypress-support/src/utils.js [hunk 10]
```diff
@@ -997,20 +997,46 @@ global.resolveAtRuntime = function (input) {
           [pattern, functionName] = pattern.split('.');
         }
 
-        // If a function name is present in the pattern, call the function with pattern content as input.
-        // Reading the pattern content from the runtime environment variable
-        const value = runtimeEnv[pattern] !== undefined ? runtimeEnv[pattern] : match;
-        return functionName && functions[functionName] ? functions[functionName](value) : value;
+        // If the input contains '->', Consdering it as an object path and extracting the value from the object
```
[the remark this answers is not in the document]
Comment (anjalimukundan) [reply cypress/support/cypress-support/src/utils.js]: handled in deeplink PR

**Review by jbigel — DISMISSED** [review]

**Review by ibrahimshuail — APPROVED** [review]

**Pull request merged**